### ISSUE TITLE
strict-boolean-expressions: add config option to disable warnings in case of enums

### ISSUE THREAD

**Issue by atsu85** [palantir/tslint#3571]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export enum StringEnum {
	VAL1 = 'VAL1',
	VAL2 = 'VAL2',
}

const varTypeIsStringEnum = StringEnum.VAL1; // type is StringEnum, but it is also string (as string values are assigned to enum constants)
if (varTypeIsStringEnum) { // ERROR: This type is not allowed in the 'if' condition because it is an enum. Allowed types are boolean, null-union, undefined-union, string, or number
}

const varTypeIsString = varTypeIsStringEnum as string;
if (varTypeIsString) { // OK, because type is cast to string
}
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"strict-boolean-expressions": [
			true,
			"allow-boolean-or-undefined",
			"allow-undefined-union",
			"allow-null-union",
			"allow-number",
			"allow-string",
			"allow-mix"
		],
		...
	},
	...
}
```

#### Actual behavior
Error is reported when using enum in boolean expression
```
ERROR: C:/tmp/demo.ts[7, 5]: This type is not allowed in the 'if' condition because it is an enum. Allowed types are boolean, null-union, undefined-union, string, or number.
```

#### Expected behavior
Would be
a) super-cool if TsLint could detect that the enum constant is defined with string value, so that the variable at runtime is string
OR
b) configuration option, smth like "allow-enum" would be added to disable warnings when using any kind of enum constants (either string-valued or normal)

so that error wouldn't be emitted
